### PR TITLE
Update GetValidatorPerformance to support validator indices

### PR DESCRIPTION
**Pull request by michaelhly**
**What type of PR is this?**
Feature

**What does this PR do? Why is it needed?**
Support for getting validator performance response with validator indices to make consistent with other RPC methods

**Which issues(s) does this PR fix?**
https://github.com/prysmaticlabs/prysm/issues/5631

### CODE REVIEW

File: beacon-chain/rpc/beacon/validators.go [hunk 1]
```diff
@@ -934,22 +934,52 @@ func (bs *Server) GetValidatorPerformance(
...
+		}
+
+		summary := validatorSummary[idx]
+		effectiveBalances = append(effectiveBalances, summary.CurrentEpochEffectiveBalance)
+		beforeTransitionBalances = append(beforeTransitionBalances, summary.BeforeEpochTransitionBalance)
+		afterTransitionBalances = append(afterTransitionBalances, summary.AfterEpochTransitionBalance)
+		inclusionSlots = append(inclusionSlots, summary.InclusionSlot)
+		inclusionDistances = append(inclusionDistances, summary.InclusionDistance)
+		correctlyVotedSource = append(correctlyVotedSource, summary.IsPrevEpochAttester)
+		correctlyVotedTarget = append(correctlyVotedTarget, summary.IsPrevEpochTargetAttester)
```
Comment (0xKiwi) [inline beacon-chain/rpc/beacon/validators.go:979]: Hello! Thanks for doing this! :) 
Theres quite a bit of duplicate code here, maybe you can make an array of all the indexes prior to getting the info from the summary?

Comment (michaelhly) [reply beacon-chain/rpc/beacon/validators.go:979]: Yes. I can do that.

**Comment by michaelhly** [comment]
@0xKiwi This is ready for review again!

File: beacon-chain/rpc/beacon/validators_test.go [hunk 2]
```diff
@@ -1948,6 +1948,197 @@ func TestGetValidatorPerformance_OK(t *testing.T) {
 	res, err := bs.GetValidatorPerformance(ctx, &ethpb.ValidatorPerformanceRequest{
 		PublicKeys: [][]byte{publicKey1[:], publicKey2[:], publicKey3[:]},
 	})
+
```
Comment (0xKiwi) [inline beacon-chain/rpc/beacon/validators_test.go:1951]:
```suggestion
```

File: beacon-chain/rpc/beacon/validators.go [hunk 3]
```diff
@@ -934,55 +934,81 @@ func (bs *Server) GetValidatorPerformance(
...
+	for _, idx := range req.Indices {
+		if !filtered[idx] {
+			validatorIndices = append(validatorIndices, idx)
+			filtered[idx] = true
+		}
+	}
+	// Depending on the indices and public keys given, results might not be sorted.
+	sort.Slice(validatorIndices, func(i, j int) bool {
+		return validatorIndices[i] < validatorIndices[j]
+	})
```
Comment (0xKiwi) [inline beacon-chain/rpc/beacon/validators.go:983]:
```suggestion
	})
	
```

**Comment by 0xKiwi** [comment]
Looks great! Thanks for doing this! :smile:

**Comment by prestonvanloon** [comment]
Also revert executable permissions on `beacon-chain/rpc/beacon/validators.go`

**Comment by michaelhly** [comment]
> ```
> WORKSPACE 100644 → 100755
> ```
> 
> Please revert making the workspace file executable.

> Also revert executable permissions on `beacon-chain/rpc/beacon/validators.go`

Whoops. Sorry about that. Reverted.

**Comment by rauljordan** [comment]
PTAL @prestonvanloon

File: beacon-chain/rpc/beacon/validators_test.go [hunk 4]
```diff
@@ -1971,6 +1971,198 @@ func TestGetValidatorPerformance_OK(t *testing.T) {
...
+		CorrectlyVotedHead:            []bool{true, true},
+		BalancesBeforeEpochTransition: []uint64{defaultBal, extraBal},
+		BalancesAfterEpochTransition:  []uint64{extraBal, extraBal + params.BeaconConfig().GweiPerEth},
+		MissingValidators:             [][]byte{publicKey1[:]},
+	}
+
+	res, err := bs.GetValidatorPerformance(ctx, &ethpb.ValidatorPerformanceRequest{
+		PublicKeys: [][]byte{publicKey1[:], publicKey3[:]}, Indices: []uint64{1, 2},
+	})
+
```
Comment (terencechain) [inline beacon-chain/rpc/beacon/validators_test.go:2157]:
```suggestion
```

File: beacon-chain/rpc/beacon/validators_test.go [hunk 5]
```diff
@@ -1971,6 +1971,198 @@ func TestGetValidatorPerformance_OK(t *testing.T) {
...
+		CorrectlyVotedHead:            []bool{true, true},
+		BalancesBeforeEpochTransition: []uint64{defaultBal, extraBal},
+		BalancesAfterEpochTransition:  []uint64{extraBal, extraBal + params.BeaconConfig().GweiPerEth},
+		MissingValidators:             [][]byte{publicKey1[:]},
+	}
+
+	res, err := bs.GetValidatorPerformance(ctx, &ethpb.ValidatorPerformanceRequest{
+		Indices: []uint64{0, 1, 2},
+	})
+
```
Comment (terencechain) [inline beacon-chain/rpc/beacon/validators_test.go:2061]:
```suggestion
```

**Pull request merged**